.github/workflows/main.yml

```diff
@@ -0,0 +1,78 @@
+name: CI
+on:
+- pull_request
+- push
+jobs:
+  tests:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        include:
+        - python-version: 3
+          env:
+            TOXENV: docs
+
+        - python-version: 3.6
+          env:
+            TOXENV: py
+        - python-version: 3.7
+          env:
+            TOXENV: py
+        - python-version: 3.8
+          env:
+            TOXENV: py
+        - python-version: 3.9
+          env:
+            TOXENV: py
+
+        - python-version: pypy-3.7
+          env:
+            TOXENV: py
+
+        - python-version: 3.9
+          env:
+            TOXENV: extra-deps
+    steps:
+      - uses: actions/checkout@v2
+      - name: Install system libraries
+        if: contains(matrix.python-version, 'pypy')
+        run: |
+          sudo apt-get update
+          sudo apt-get install libxml2-dev libxslt-dev
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install --upgrade tox codecov
+      - name: Run tests
+        env: ${{ matrix.env }}
+        run: tox
+      - name: Publish coverage data
+        uses: codecov/codecov-action@v1
+  publish:
```

Comment:
**suggestion**: What do you think about creating a separated workflow for publishing (like the one we have on [scrapy](https://github.com/scrapy/scrapy/blob/master/.github/workflows/publish.yml) or [spidermon](https://github.com/scrapinghub/spidermon/blob/master/.github/workflows/publish.yml)?